llvm/test/CodeGen/AMDGPU/default-flat-work-group-size-overrides-waves-per-eu.ll

```diff
@@ -1,61 +0,0 @@
-; RUN: opt -S -mtriple=amdgcn-unknown-unknown -mcpu=tahiti -passes=amdgpu-promote-alloca -disable-promote-alloca-to-vector < %s | FileCheck %s
-
-; Both of these kernels have the same value for
-; amdgpu-flat-work-group-size, except one explicitly sets it. This is
-; a program visible property which should always take precedence over
-; the amdgpu-waves-per-eu optimization hint.
-;
-; The range is incompatible with the amdgpu-waves-per-eu value, so the
-; flat work group size should take precedence implying a requirement
-; to support 1024 size workgroups (which exceeds the available LDS
-; amount).
-
-; CHECK-NOT: @no_flat_workgroup_size.stack
-; CHECK-NOT: @explicit_default_workgroup_size.stack
-
-; CHECK-LABEL: @no_flat_workgroup_size(
-; CHECK: alloca [5 x i32]
```

Comment:
I don't understand why this test was deleted, flat-workgroup-size still needs to win

Reply:
I renamed this test file to `waves-per-eu-hints-lower-occupancy-target.ll` and changed the functions' attributes because the test's statement/name no longer made sense with the new design I think.

AFAIU flat workgroup size only limits the maximum occupancy (not the minimum, since e.g., using all registers with any flat workgroup size will always lead to an occupancy of 1), but I think waves-per-eu should be able to hint that an arbitrarily lower occupancy (here 1) is desired. This doesn't override the ABI requirement, but does restrict the target range over what we would get by default without waves-per-eu.